db/migrate/20240705134114_clear_orphaned_draft_upload_references2.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class ClearOrphanedDraftUploadReferences2 < ActiveRecord::Migration[7.1]
+  def up
+    execute <<~SQL
+      DELETE
+      FROM
+        "upload_references"
+      WHERE
+        "upload_references"."target_type" = 'Draft' AND
+        "upload_references"."target_id" NOT IN (
+          SELECT "drafts"."id" FROM "drafts"
+        )
```

Comment:
`NOT IN` can sometimes result in poor performing query plans. Perhaps we can just do a `LEFT JOIN` on the `drafts` table?

Reply:
It depends a lot on the use case, but in this case we cannot use a JOIN with a `DELETE` query. I could move more to the subquery or a CTE, but I do not think that would be any better.

We have ran this migration in the past (exact same migration has been merged a few months ago) and we had no problems.

Reply:
> We have ran this migration in the past (exact same migration has been merged a few months ago) and we had no problems.

👍 We can always rewrite the migration if it causes problems.